src/java.base/share/classes/java/lang/invoke/LambdaFormEditor.java

```diff
@@ -186,43 +191,44 @@ static TransformKey of(byte b0, int b1, int b2, int b3) {
             else
                 return new TransformKey(fullBytes(b0, b1, b2, b3));
         }
-        private static final byte[] NO_BYTES = {};
         static TransformKey of(byte kind, int... b123) {
-            return ofBothArrays(kind, b123, NO_BYTES);
+            long packedBytes = packedBytes(kind, b123);
+            if (packedBytes != 0) {
+                return new TransformKey(packedBytes);
+            }
+            byte[] fullBytes = new byte[b123.length + 1];
+            fullBytes[0] = kind;
+            for (int i = 0; i < b123.length; i++) {
+                fullBytes[i + 1] = TransformKey.bval(b123[i]);
+            }
+            return new TransformKey(fullBytes);
         }
-
-        static TransformKey of(byte kind, int b1, int[] b23456) {
-            byte[] fullBytes = new byte[b23456.length + 2];
+        static TransformKey of(byte kind, int b1, int... b234) {
+            long packedBytes = packedBytes(kind, b1, b234);
+            if (packedBytes != 0) {
+                return new TransformKey(packedBytes);
+            }
+            byte[] fullBytes = new byte[b234.length + 2];
             fullBytes[0] = kind;
             fullBytes[1] = bval(b1);
-            for (int i = 0; i < b23456.length; i++) {
-                fullBytes[i + 2] = TransformKey.bval(b23456[i]);
+            for (int i = 0; i < b234.length; i++) {
+                fullBytes[i + 2] = TransformKey.bval(b234[i]);
             }
-            long packedBytes = packedBytes(fullBytes);
-            if (packedBytes != 0)
-                return new TransformKey(packedBytes);
-            else
-                return new TransformKey(fullBytes);
-        }
-
-        static TransformKey of(byte kind, int b1, int b2, byte[] b345) {
-            return ofBothArrays(kind, new int[]{ b1, b2 }, b345);
+            return new TransformKey(fullBytes);
         }
-        private static TransformKey ofBothArrays(byte kind, int[] b123, byte[] b456) {
-            byte[] fullBytes = new byte[1 + b123.length + b456.length];
-            int i = 0;
-            fullBytes[i++] = bval(kind);
-            for (int bv : b123) {
-                fullBytes[i++] = bval(bv);
+        static TransformKey of(byte kind, int b1, int b2, int[] b3456) {
```

Comment:
Nit: I can't figure out if `b3456` intends to be a different convention than the others `b123` and `b234`.  I would expect something like this:

```suggestion
        static TransformKey of(byte kind, int b1, int b2, int... b345) {
```